src/main/java/com/hashedin/bitly/ShortCodeMappingService.java

```diff
@@ -0,0 +1,63 @@
+package com.hashedin.bitly;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ShortCodeMappingService {
+
+	private final Map<String, URI> mapping;
+	
+	public ShortCodeMappingService(@Value("classpath:mapping.properties") Resource mappingResource) throws IOException {
+		this.mapping = loadMapping(mappingResource);
+	}
```

Comment:
Notice we use the Resource class instead of a java.io.File or an InputStream or a Reader. Resource is a spring class that makes it easy to read from a variety of places - including classpath, URL and so on. It also makes unit testing a lot easy.

In a more mature application, we would make the file name also parametric.